framework/src/postprocessors/FindValueOnLine.C

```diff
@@ -90,7 +90,7 @@ FindValueOnLine::execute()
 
   bool found_it = false;
   Real value = 0;
-  for (auto i = decltype(_depth)(0); i < _depth; ++i)
+  for (unsigned int i = 0; i < _depth; ++i)
```

Comment:
Interesting, so we used that syntax precisely to get around the `const` problem (as opposed to `for (decltype(_depth) i = 0 ...`. So I guess C++ really can't give us a good way to get the right type while removing const?

Reply:
yeah. I think the const on the member variable has value, but then removing the const after `decltype` is IMO much worse then just doing plain old `unsigned int i = ...`.

Reply:
Agreed, but we've seen several places where `unsigned int` bites us as we grow beyond the 32-bit sizes. I wish there was a clean syntax that let us loop over the current size type.

I'm fine with this for now until we figure out something better.